weld-python/weld/grizzly/core/strings.py

```diff
@@ -0,0 +1,263 @@
+
+    def to_pandas(self):
+        """
+        Convert an array of strings to a Pandas series.
+
+        We provide a specialized implementation of `to_pandas` here that will perform UTF-8 decoding
+        of the raw bytestrings that Grizzly series operate over.
+
+        Examples
+        --------
+        >>> from weld.grizzly import GrizzlySeries
+        >>> x = GrizzlySeries(["Welcome", "to", "Grizzly!"])
+        >>> x
+        0     b'Welcome'
+        1          b'to'
+        2    b'Grizzly!'
+        dtype: bytes64
+        >>> x.str.to_pandas()
+        0     Welcome
+        1          to
+        2    Grizzly!
+        dtype: object
+
+        """
+        return self.series.evaluate().to_pandas().str.decode("utf-8")
+
+    def _apply(self, func, *args, return_weld_elem_type=None):
+        """
+        Apply the given weldfunc to `self.series` and return a new GrizzlySeries.
+
+        If the return type of the result is not a string GrizzlySeries, pass
+        'return_weld_elem_type' to specify the element type of the result.
+
+        """
+        output_type = self.series.output_type if return_weld_elem_type is None else WeldVec(return_weld_elem_type)
+        dtype = 'S' if return_weld_elem_type is None else wenp.weld_type_to_dtype(return_weld_elem_type)
+        lazy = func(self.series.weld_value_, *args)(output_type, self.constructor._decoder)
+        return (self.constructor)(lazy, dtype=dtype)
+
+    def lower(self):
+        """
+        Lowercase strings.
+
+        Examples
+        --------
+        >>> from weld.grizzly import GrizzlySeries
+        >>> x = GrizzlySeries(["HELLO", "WorLD"])
+        >>> x.str.lower().str.to_pandas()
+        0    hello
+        1    world
+        dtype: object
+
+        """
+        return self._apply(weldstr.lower)
+
+    def upper(self):
+        """
+        Uppercase strings.
+
+        Examples
+        --------
+        >>> from weld.grizzly import GrizzlySeries
+        >>> x = GrizzlySeries(["hello", "WorlD"])
+        >>> x.str.upper().str.to_pandas()
+        0    HELLO
+        1    WORLD
+        dtype: object
+
+        """
+        return self._apply(weldstr.upper)
+
+    def capitalize(self):
+        """
+        Capitalize the first character in each string.
+
+        Examples
+        --------
+        >>> from weld.grizzly import GrizzlySeries
+        >>> x = GrizzlySeries(["hello", "worlD"])
+        >>> x.str.capitalize().str.to_pandas()
+        0    Hello
+        1    World
+        dtype: object
+
+        """
+        return self._apply(weldstr.capitalize)
+
+    def get(self, index):
+        """
+        Get the character at index 'i' from each string. If 'index' is greater than
+        the string length, this returns an empty string. If 'index' is less than 0,
+        this wraps around, using Python's indexing behavior.
+
+        Examples
+        --------
+        >>> from weld.grizzly import GrizzlySeries
```

Comment:
@deepakn94 any idea how to fix this? Seems like there should be a way to add a common prelude to doctests, but didn't find how to do it...

Reply:
See weld-python/weld/conftest.py for how to do this for future reference (thanks @paroma!)

Reply:
Interesting -- good to know for the future!